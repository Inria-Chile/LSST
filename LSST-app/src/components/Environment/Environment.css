.environment-container table {
    width: calc(100% - 10px);
    height: 100%;
}

.environment-container .table-wrapper {
    background: var(--theme-div-background);
    padding: 10px 20px;
    margin: 0px 5px;
}

.environment-container table tbody tr{
    border-bottom: 2px solid var(--theme-active-slider);
}

.environment-container table tbody td{    
    padding: 10px 20px;
}

td.telemetry-name {
    font-weight: bold;
    width: 170px;
}

.tick text {
    fill: #dddddd;
}

.environment-container .hidden svg {
    max-height: 0;
}

.environment-container  svg {
    max-height: 300px;
    transition: max-height 0.5s ease-in-out;
}

.environment-container .plot-div {
}

.environment-container .plot-column {
}

.environment-container tr {
    vertical-align: top;
    cursor: pointer;
}

.environment-container td {
    vertical-align: top;
    cursor: pointer;

}