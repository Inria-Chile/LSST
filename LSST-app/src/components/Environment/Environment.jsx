import React, { Component } from 'react';
// import ReactDOM from 'react-dom';
import './Environment.css';
import DraggableTitle from '../Utils/DraggableTitle';
import openSocket from 'socket.io-client';

class Environment extends Component {

    constructor(props) {
        super(props);
        this.state = {
            temperature: null,
            humidity: null,
            pressure: null,
            temperatureArray: [],
            humidityArray: [],
            pressureArray: [],
        }
        this.socket = openSocket(window.location.origin + '/weather');      
        console.log('Environment\'s socket listening', this.socket.on('Weather', msg => this.receiveEnvironmentData(msg)));
    }

    receiveEnvironmentData = (msg) => {
        console.log('receiveEnvironmentData', msg);
        // if(msg.position_actual){
        //     this.setState({
        //         louversAperture: msg.position_actual
        //     });
        // }
    }

    componentDidMount() {
        this.setState({
            temperature: 32,
            humidity: 57,
            pressure: 770,
            temperatureArray: [],
            humidityArray: [],
            pressureArray: [],
        })
    }

    componentWillReceiveProps() {
        let maxArrayLength = 100;
        this.state.temperatureArray.push(this.props.temperature);
        this.state.humidityArray.push(this.props.humidity);
        this.state.pressureArray.push(this.props.pressure);
        
        if (this.state.temperatureArray.length > maxArrayLength)
            this.state.temperatureArray.shift();
        if (this.state.humidityArray.length > maxArrayLength)
            this.state.humidityArray.shift();
        if (this.state.pressureArray.length > maxArrayLength)
            this.state.pressureArray.shift();
    }

    render() {
        let temperature = this.props.temperature;
        console.log("TEMPERATURA",temperature);
        return (
            <div className="environment-container">
                <DraggableTitle title='Environment'/>
                <div className='table-wrapper'>
                    <table>
                        <tbody>
                            <tr>
                                <td className='telemetry-name'>Temperature</td>
                                {
                                    temperature ?
                                    <td>{(temperature*9/5+32).toFixed(1)}ºF / {temperature.toFixed(1)}ºC</td>
                                    :
                                    <td>Unavailable</td>
                                }
                            </tr>
                            <tr>
                                <td className='telemetry-name'>Humidity</td>
                                {
                                    this.state.humidity ?
                                    <td>{this.state.humidity}%</td>
                                    :
                                    <td>Unavailable</td>
                                }
                            </tr>
                            <tr>
                                <td className='telemetry-name'>Pressure</td>
                                {
                                    this.state.pressure ?
                                    <td>{this.state.pressure} mmHg</td>
                                    :
                                    <td>Unavailable</td>
                                }
                            </tr>
                        </tbody>
                    </table>                    
                </div>
            </div>
        );
    }
}

export default Environment;
