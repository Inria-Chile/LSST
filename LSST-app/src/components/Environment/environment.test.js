import React from 'react';
import Environment from './Environment';
import renderer from 'react-test-renderer';


it('renders correctly', () => {
  const tree = renderer
    .create(<Environment />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

