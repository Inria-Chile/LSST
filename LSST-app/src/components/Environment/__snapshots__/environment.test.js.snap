// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="environment-container"
>
  <div
    className="draggable-title "
  >
    <h2>
      Environment
    </h2>
    <div
      className="move-button"
    >
      <svg
        fill="currentColor"
        height={30}
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width={30}
      >
        <g>
          <path
            d="m29.5 13.8c-0.6-0.6-1.7-0.6-2.3 0s-0.7 1.7 0 2.4l2.1 2.1h-7.6v-7.6l2.1 2.1c0.3 0.4 0.8 0.5 1.2 0.5s0.9-0.1 1.2-0.5c0.6-0.6 0.6-1.7 0-2.3l-6.2-6.2-6.2 6.2c-0.6 0.6-0.6 1.7 0 2.3s1.7 0.7 2.4 0l2.1-2.1v7.6h-7.6l2.1-2.1c0.7-0.7 0.7-1.7 0-2.4s-1.7-0.6-2.3 0l-6.2 6.2 6.2 6.2c0.3 0.3 0.7 0.5 1.2 0.5s0.8-0.2 1.1-0.5c0.7-0.7 0.7-1.7 0-2.4l-2.1-2.1h7.6v7.6l-2.1-2.1c-0.7-0.7-1.7-0.7-2.4 0s-0.6 1.7 0 2.3l6.2 6.2 6.2-6.2c0.6-0.6 0.6-1.7 0-2.3s-1.7-0.7-2.4 0l-2.1 2.1v-7.6h7.6l-2.1 2.1c-0.7 0.7-0.7 1.7 0 2.4 0.3 0.3 0.7 0.5 1.1 0.5s0.9-0.2 1.2-0.5l6.2-6.2-6.2-6.2z"
          />
        </g>
      </svg>
    </div>
    
  </div>
  <div
    className="table-wrapper"
  >
    <table>
      <tbody>
        <tr>
          <td
            className="telemetry-name"
          >
            Temperature
          </td>
          <td>
            89.6
            ºF / 
            32.0
            ºC
          </td>
        </tr>
        <tr>
          <td
            className="telemetry-name"
          >
            Humidity
          </td>
          <td>
            57
            %
          </td>
        </tr>
        <tr>
          <td
            className="telemetry-name"
          >
            Pressure
          </td>
          <td>
            770
             mmHg
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
